### PR TITLE
feat(CLI): Add push

### LINKED ISSUE #201: feat(CLI): Support push command
## Description

Push the envd environment to the docker registry.

### PR DESCRIPTION
**Pull request by gaocegege**
Fix #201 

Ref #510 

```
   To build an image using build.envd:
     $ envd build
   To build and push the image to a registry:
     $ envd build --output type=image,name=docker.io/username/image,push=true
```

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

File: pkg/builder/util.go [hunk 1]
```diff
@@ -63,33 +68,104 @@ func DefaultPathEnv(os string) string {
 	return DefaultPathEnvUnix
 }
 
-// parseOutput parses the output string and returns the output type and destination.
-func parseOutput(output string) (string, string, error) {
-	if output == "" {
-		return "", "", nil
+// ParseOutput parses --output
```
Comment (rudeigerc) [inline pkg/builder/util.go:71]: s/ParseOutput/parseOutput/

**Review by rudeigerc — APPROVED** [review]
LGTM

By the way, it would be better for us to explain the available keys of `--output`, or refer to the documentation of `buildkit`.

**Comment by gaocegege** [comment]
Yep, I think we can document it.

**Comment by gaocegege** [comment]
https://coveralls.io/github/tensorchord/envd coveralls got server down.

Let's wait

**Review by VoVAllen — APPROVED** [review]

File: pkg/builder/util.go [hunk 2]
```diff
@@ -63,33 +68,104 @@ func DefaultPathEnv(os string) string {
...
+// Refer to https://github.com/moby/buildkit/blob/master/cmd/buildctl/build/output.go#L56
+func parseOutput(exports string) ([]client.ExportEntry, error) {
+	var entries []client.ExportEntry
+	if exports == "" {
+		return entries, nil
 	}
 
-	// Example: type=tar,dest=path
-	matched, err := regexp.Match(`^type=[\s\S]+,dest=[\s\S]+$`, []byte(output))
+	e, err := parseOutputCSV(exports)
```
Comment (VoVAllen) [inline pkg/builder/util.go:79]: What's csv for?

**Review by VoVAllen — COMMENTED** [review]
Overall LGTM

**Comment by VoVAllen** [comment]
Also this didn't solve #510, right?

File: pkg/builder/util.go (hunk 2 above)
Comment (gaocegege) [reply pkg/builder/util.go:79]: The output `type=tar,image=imagename,push=true` is CSV-format.

**Comment by gaocegege** [comment]
> Also this didn't solve #510, right?

We need to discuss it with users to decide if it works.

After the PR, you can push the image in GitHub actions.

**Pull request merged**